app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -173,16 +189,18 @@ public void onPairingButtonClicked(final View view) {
                 break;
             case R.id.pairing_menu_nfc:
                 intent.setComponent(new ComponentName(this, NFCActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+                this.startActivity(intent);
                 break;
             case R.id.pairing_menu_qr:
                 intent.setComponent(new ComponentName(this, QRExchangeKeyActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+                this.startActivity(intent);
                 break;
             default:
                 LOGGER.error("Unknown pairing button clicked");
                 throw new IllegalArgumentException("Only existing buttons can be clicked");
         }
-
-        this.startActivity(intent);
```

Comment:
No this was on purpose. Case "R.id.pairing_menu_bluetooth:" was changed to give a popup when Bluetooth was not enabled. After this is enabled via the popup it will go to the desired activity.